### PR TITLE
Add Leaderboard Model

### PR DESCRIPTION
**Pull request by zysim**
This adds `Leaderboard`. Quite a few things left to do, as listed below, which speaking of, I'd like to ask which items should be included in this PR, and which ones should be left for another.

### Things Done

- Leaderboard model, context, and controller
- Service registration in `Program.cs`
- Creation of and finding leaderboards (single and multiple IDs)

### Todo

- [x] More tests

### Future Work

- Put and delete requests (creation request done)
- Authentication + authorisation (e.g. how only admins should be able to create a leaderboard)
- Relationship definitions with other models

### CODE REVIEW

**Comment by zysim** [comment]
Not writing a test for creating a leaderboard. With the way `CreateLeaderboard()`'s written, it'll be kinda pointless to write one lmao

**Comment by zysim** [comment]
I think I'll keep this PR to these changes. Everything else in the MVP, I'll make more PRs for em.

**Comment by RageCage64** [comment]
We are still getting murdered by tabs. 😢 
Do you have the editorconfig plugin installed currently? Maybe that will help

File: LeaderboardBackend.Test/Controllers/LeaderboardsControllerTest.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    {
+		_leaderboardServiceMock = new Mock<ILeaderboardService>();
+
+		_controller = new LeaderboardsController(
+			_leaderboardServiceMock.Object
+		);
+    }
+
+    [Test]
+    public async Task TestGetLeaderboardNotFoundIfLeaderboardNotFound()
```
Comment (RageCage64) [inline LeaderboardBackend.Test/Controllers/LeaderboardsControllerTest.cs:30]: [This test naming style](https://github.com/leaderboardsgg/leaderboard-backend-poc/blob/main/LeaderboardBackend.Test/Controllers/UsersControllerTest.cs#L46) is what I tried in the UsersController tests, and it seems pretty readable. Would it be okay to swap these over to that or is this way better?

File: LeaderboardBackend/Services/ILeaderboardService.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+using LeaderboardBackend.Models;
+
+namespace LeaderboardBackend.Services
+{
+    public interface ILeaderboardService
+    {
+        Task<Leaderboard?> GetLeaderboard(long id);
+
+        Task<List<Leaderboard>> GetLeaderboards(long[] ?ids = null);
```
Comment (RageCage64) [inline LeaderboardBackend/Services/ILeaderboardService.cs:9]: nit: slight inconsistency in the line break choices

File: LeaderboardBackend/Services/LeaderboardService.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            if (ids == null) {
+                return await _leaderboardContext.Leaderboards.ToListAsync();
+            } else {
+                return await _leaderboardContext.Leaderboards.Where(l => ids.Contains(l.Id)).ToListAsync();
+            }
+        }
+
+        public async Task CreateLeaderboard(Leaderboard leaderboard)
+        {
+            _leaderboardContext.Leaderboards.Add(leaderboard);
```
Comment (RageCage64) [inline LeaderboardBackend/Services/LeaderboardService.cs:34]: Not a real review comment but writing it down so it's in our heads: we should start adding validation to the models before we create them. We could call some `leaderboard.Validate()` method before creating them in this spot, would be a pretty good spot to enforce it.

**Review by RageCage64 — APPROVED** [review]
Try with editorconfig if you haven't already, we'll try and figure out what is going on with the tabbing here.

File: LeaderboardBackend.Test/Controllers/LeaderboardsControllerTest.cs (hunk 1 above)
Comment (zysim) [reply LeaderboardBackend.Test/Controllers/LeaderboardsControllerTest.cs:30]: Let's go with that style; definitely more readable.

File: LeaderboardBackend/Services/ILeaderboardService.cs (hunk 2 above)
Comment (zysim) [reply LeaderboardBackend/Services/ILeaderboardService.cs:9]: Don't think I can see this in my editor.. would you mind fixing this?

Comment (RageCage64) [reply LeaderboardBackend/Services/ILeaderboardService.cs]: We can leave it and figure out what happened when this is merged.

**Review by RageCage64 — APPROVED** [review]

**Comment by zysim** [comment]
Aight gonna rebase and then mergeeee

File: LeaderboardBackend/Services/LeaderboardService.cs (hunk 3 above)
Comment (zysim) [reply LeaderboardBackend/Services/LeaderboardService.cs]: Made issue from this comment :+1:

**Pull request merged**